exonum/src/encoding/protobuf/mod.rs

```diff
@@ -46,21 +46,20 @@ pub trait ProtobufConvert: Sized {
     fn to_pb(&self) -> Self::ProtoStruct;
 
     /// ProtoStruct -> Struct
-    fn from_pb(pb: Self::ProtoStruct) -> Result<Self, ()>;
+    fn from_pb(pb: Self::ProtoStruct) -> Result<Self, Error>;
 }
 
 impl<T> BinaryForm for T
 where
     T: Message,
 {
     fn encode(&self) -> Result<Vec<u8>, Error> {
-        self.write_to_bytes().map_err(|e| Error::Other(Box::new(e)))
+        self.write_to_bytes().map_err(|e| e.into())
```

Comment:
You can just write `map_err(Error::from)`

Reply:
Ok, I already received similar comments but I preferred this form because you do not name type. But I change it.